src/wp-includes/formatting.php

```diff
@@ -3914,10 +3914,22 @@ function wp_trim_excerpt( $text = '', $post = null ) {
 		$text = strip_shortcodes( $text );
 		$text = excerpt_remove_blocks( $text );
 
+		/*
+		 * Temporarily unhook wp_filter_content_tags() since any tags
+		 * within the excerpt are stripped out. Modifying the tags here
+		 * is wasteful and can lead to bugs in the image counting logic.
+		 */
+		$filter_removed = remove_filter( 'the_content', 'wp_filter_content_tags' );
```

Comment:
```suggestion
                 $priority = has_filter(  'the_content', 'wp_filter_content_tags' );
                 $filter_removed = false;
                 if(  $priority ){
		     $filter_removed = remove_filter( 'the_content', 'wp_filter_content_tags', $priority );
		}
```

@costdev Is this what you had in mind?

Reply:
@spacedmonkey Similar to that yeah.

```php
$filter_priority = has_filter( 'the_content', 'wp_filter_content_tags' );

if ( is_int( $filter_priority ) ) { // Or false !== $filter_priority
    remove_filter( 'the_content', 'wp_filter_content_tags' );
}
```

Later:
```php
if ( is_int( $filter_priority ) ) { // Or false !== $filter_priority
    add_filter( 'the_content', 'wp_filter_content_tags', $filter_priority );
}
```

- `has_filter()` will return `int|false` in this context, which could even be `0`, so a strict check is needed.
- The call to `remove_filter()` is dependent on the result of `has_filter()`, so we don't need to store the result of `remove_filter()` - we know it'll be `true`.

Reply:
See my reply in https://github.com/WordPress/wordpress-develop/pull/4439#discussion_r1198081475. I'm not strongly opposed, but it seems unnecessary overhead to me.

Reply:
It feel a bit of a edge case for us to worry about. If a developer has double hooked this on, they may have a good reason.

Let's flag this in the dev note.

Reply:
A couple cases of prior art in core:

The `wp_pre_kses_block_attributes()` function:

https://github.com/WordPress/wordpress-develop/blob/b1f2a760a283c24511623e1c13421f20e27fed88/src/wp-includes/formatting.php#L5162-L5164

The `WP_Widget_Text::widget()` method:

https://github.com/WordPress/wordpress-develop/blob/472c9f242ba6d9141bdb4d239aa9c725abb53214/src/wp-includes/widgets/class-wp-widget-text.php#L249-L253

https://github.com/WordPress/wordpress-develop/blob/472c9f242ba6d9141bdb4d239aa9c725abb53214/src/wp-includes/widgets/class-wp-widget-text.php#L321-L324

Reply:
@westonruter No alignment on how to handle the priority between those cases either, it seems 😆 

I personally think checking for the priority is overkill, given it is controlled by core and not intended to be changed by plugin developers. Removing one of those crucial `the_content` filters or changing their order has a high chance to break other plugins and therefore is not recommended.